pennylane/devices/qutrit_mixed/sampling.py

```diff
@@ -0,0 +1,367 @@
+    obs_terms = mp.obs if isinstance(mp.obs, Sum) else mp.obs.terms()[1]
+
+    def _sum_for_single_shot(s):
+        results = []
+        for term in obs_terms:
+            results.append(
+                measure_with_samples(
+                    ExpectationMP(term),
+                    state,
+                    s,
+                    is_state_batched=is_state_batched,
+                    rng=rng,
+                    prng_key=prng_key,
+                )
+            )
+
+        if isinstance(mp.obs, Hamiltonian):
+            # If Hamiltonian apply coefficients
+            return sum((c * res for c, res in zip(mp.obs.terms()[0], results)))
+        return sum(results)
+
+    if shots.has_partitioned_shots:
+        return tuple(_sum_for_single_shot(type(shots)(s)) for s in shots)
+    return _sum_for_single_shot(shots)
+
+
+def _sample_state_jax(
+    state,
+    shots: int,
+    prng_key,
+    is_state_batched: bool = False,
+    wires=None,
+) -> np.ndarray:
+    """Returns a series of samples of a state for the JAX interface based on the PRNG.
+
+    Args:
+        state (array[complex]): A state vector to be sampled
+        shots (int): The number of samples to take
+        prng_key (jax.random.PRNGKey): A``jax.random.PRNGKey``. This is
+            the key to the JAX pseudo random number generator.
+        is_state_batched (bool): whether the state is batched or not
+        wires (Sequence[int]): The wires to sample
+
+    Returns:
+        ndarray[int]: Sample values of the shape (shots, num_wires)
+    """
+    # pylint: disable=import-outside-toplevel
+    import jax
+    import jax.numpy as jnp
+
+    key = prng_key
+
+    total_indices = get_num_wires(state, is_state_batched)
+    state_wires = qml.wires.Wires(range(total_indices))
+
+    wires_to_sample = wires or state_wires
+    num_wires = len(wires_to_sample)
+    basis_states = np.arange(QUDIT_DIM**num_wires)
+
+    with qml.queuing.QueuingManager.stop_recording():
+        probs = measure(qml.probs(wires=wires_to_sample), state, is_state_batched)
+
+    if is_state_batched:
+        # Produce separate keys for each of the probabilities along the broadcasted axis
+        keys = []
+        for _ in state:
+            key, subkey = jax.random.split(key)
+            keys.append(subkey)
+        samples = jnp.array(
+            [
+                jax.random.choice(_key, basis_states, shape=(shots,), p=prob)
+                for _key, prob in zip(keys, probs)
+            ]
+        )
+    else:
+        samples = jax.random.choice(key, basis_states, shape=(shots,), p=probs)
+
+    res = np.zeros(samples.shape + (num_wires,), dtype=np.int64)
+    for i in range(num_wires):
+        res[..., -(i + 1)] = (samples // (QUDIT_DIM**i)) % QUDIT_DIM
+    return res
+
+
+def sample_state(
+    state,
+    shots: int,
+    is_state_batched: bool = False,
+    wires=None,
+    rng=None,
+    prng_key=None,
+) -> np.ndarray:
+    """Returns a series of samples of a state.
```

Comment:
```suggestion
    """Returns a series of computational basis samples of a state.
```
Slightly adjust wording to reflect what gets returned; feel free to adjust if this doesn't make sense